openwillis/measures/audio/speech_phonation.py

```diff
@@ -0,0 +1,217 @@
+    df = df[[measures['phonation_type']] + [col for col in df.columns if col != measures['phonation_type']]]
+    df[measures['duration']] = duration
+
+    # remove unused columns
+    ## pause related measures
+    for col in [measures['spir'], measures['pause_meddur'], measures['pause_maddur'], measures['silence_ratio']]:
+        if col in df.columns:
+            df = df.drop(col, axis=1)
+
+    return df
+
+def phonations_acoustics(audio_path, transcript_json, speaker_label=''):
+    """
+    ------------------------------------------------------------------------------------------------------
+
+    This function extracts phonation acoustics from the audio signal based on the transcripted information,
+     then computes advanced vocal acoustics measures.
+
+    Parameters:
+    ...........
+    audio_path : str
+        Path to the input audio file.
+    transcript_json : json
+        Speech transcription json response.
+    speaker_label : str
+        Speaker label.
+
+    Returns:
+    ...........
+    phonations_df : pandas.DataFrame
+        A pandas dataframe containing the phonation-level measures.
+    summ_df : pandas.DataFrame
+        A pandas dataframe containing the summarized phonation acoustics measures.
+
+    ------------------------------------------------------------------------------------------------------
+    """
+
+    phonations_df, summ_df = pd.DataFrame(), pd.DataFrame()
+    measure_path = os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), './config/acoustic.json'))
+    file = open(measure_path)
+    measures = json.load(file)
+
+    try:
+        if not os.path.exists(audio_path) or not transcript_json:
+            raise Exception('Audio file or transcript json not found')
+
+        # extract phonation segments
+        phonation_dict = phonation_extraction(audio_path, transcript_json, speaker_label)
+        if not phonation_dict:
+            return phonations_df, summ_df
+
+        # save phonation segments
+        temp_dir = tempfile.mkdtemp()
+        to_audio(audio_path, phonation_dict, temp_dir)
+
+        for file in os.listdir(temp_dir):
+            # standardize volume level
+            audio_signal = AudioSegment.from_file(file = os.path.join(temp_dir, file), format = "wav")
+            audio_signal = sutil.volume_normalization(audio_signal, -20)
+            audio_signal.export(os.path.join(temp_dir, file), format="wav")
+
+            # compute advanced vocal acoustics measures
+            _, df = vocal_acoustics(os.path.join(temp_dir, file), option='advanced')
+            df = clean_acoustic_df(df, file, audio_signal.duration_seconds, measures)
+
+            phonations_df = pd.concat([phonations_df, df])
+
+        # summarize the phonation acoustics into dfs
+        summ_df = phonations_df.groupby(measures['phonation_type']).mean().reset_index()
+        # clear the temp directory
+        for file in os.listdir(temp_dir):
```

Comment:
instead of for loop we can use shutil.rmtree to delete all the files; just read on this in terms of performance